Adding saved searches documentation (#436)
openshift/openshift-origin-design: designs/administrator/future-openshift/saved-searches/index.md

---
parent: Administrator
version: 4.x
---

# Saved filters

The Search page will be updated to allow users to save searches.

## General flow

![1](img/saved1.png)

> * If no filters have been saved, the **Saved filters** dropdown will have a small empty state.

Reviewer: This is just an opinion that relates to the Search page in general, but I sort of feel like this pages empty state should  be actionable to get something there in a hurry, especially since the "Resources" dropdown is ever so slightly more mixed in now. I think the MCM team showed that they list out the saved filters as tiles in the empty state so you can just click one and see the matching resources? Maybe if no filters are saved have common suggested filters like "Show all pods" or something? This is just a random thought but figured I'd float out there.

Reviewer 2: Small nit. but maybe clearer if empty state says something like "Create and save filters to access them here".

Author: I like that copy @dtaylor113, makes sense to me. And yeah that's a good point @itsptk . I'm curious if PF has any empty states that do anything like that – it would be kind of cool to show a set of default filters users can easily click.

![2](img/saved2.png)
* Users can save filters by clicking the **Save filter** button at the end of the filter chip row.

> ![3](img/saved3.png)

Reviewer 3: nit: it could be nice to call this Save search filter
same for the description text: Save the current search filter...

Author: Hm, if you don't feel strongly on this, I may prefer to leave it as "Save filter". My reasoning is that the page is called Search so technically you are saving the "Search filter" but may be confusing because "Save search" sounds like a similar function. In that way I wonder if "Save search filter" clarifies or confuses. I imagine we wouldn't name this "Save Pod filter" if it appeared on the Pods page. What do you think?

* Users will be able to enter a name for the saved filter in a modal.

![4](img/saved4.png)
* The **Saved filters** dropdown will update to show the saved filter name since it is the currently applied filter.

![5](img/saved5.png)

> * Users can click the **Saved filters** dropdown to see all their saved searches.

Reviewer 3: ```suggestion
* Users can click the **Saved filters** dropdown to see all their saved search filters.
```

> * Through the dropdown, users may also choose to edit their saved searches.

Reviewer 3: ```suggestion
* Through the dropdown, users may also choose to edit their saved search filters.
```

![6](img/saved6.png)
* In the **Edit saved filters** modal, users are able to edit the name of any previously saved filter.

> ![7](img/saved7.png)

Reviewer 3: that checkmark should be blue right?

Reviewer 3: it also looks like PF would implement this with the input stretching the full width to the right to meet the edit icons

Reviewer 3: @beanh66 do you know if there is a convention update planned to focus on when action buttons should be disabled or not? I don't see one now and I think this has always been a topic that isn't fully settled - if a button should be disabled OR enabled that triggers an alert with a description of what needs to be fixed before the user can proceed. Maybe I'm overthinking it?

Author: Updated based on your comments @matthewcarleton ! And yeah, I think we've had this conversation in past and settled on what @itsptk described above – disable if it's clear what you need to fix, but if it's out of view or otherwise hard to know, enabled so they can get the appropriate error.

* Using [inline edit](https://www.patternfly.org/v4/documentation/core/components/inlineedit), users can update and save a new filter name.
* While the inline edit action is activated, the modal's **Save** button should be disabled.

![8](img/saved8.png)

> * Users can also choose to remove a saved search by clicking the minus icon.

Reviewer 2: I assume if the active filter is removed, page reverts back same as when one clicks on 'Clear Filter'.

Probably beyond this design, but I notice screenshots have the '3 selected' in the toolbar.  What are the thoughts around preserving the selected rows IF they are also in a newly selected saved filter?  I imagine it would be up to the app.   
Similar question, probably app specific, but I imagine the saved filters will be local to the particular browser (ie. saved in browser local storage/cookies).  Probably up to the app if they can save filters on server by user etc..

Author: That's a good question. @gdoyle1 I know you've done some work around bulk actions and selecting items from a list. What do you imagine would happen in the case above? Should we reset the selection if the user changes the filters? Or still filter down the list and update – for example they have 4 selected, then apply a filter, and now only have 2 selected.

![9](img/saved9.png)
* Users can save the updates they made to their saved filters list by clicking the modal's **Save** button.

## Additional behavior
* If a user selects a saved filter and then adds another filter or resource, the **Save filter** button will reappear, allowing the user to save a new filter. The **Saved filter** dropdown will return to its unselected state.
* If a user selects a saved filter and then hits **Clear filters**, all the chips are removed and the **Saved filter** dropdown will return to its unselected state.

> 

Reviewer 2: Hi, what happens if user creates/saves a filter using a name of an already existing filter?  Is it overwritten?  If so, is there a warning?  If not, do we do we name it something like 'existing-filter(1)'?

Author: That's a good question! I just did this for projects, and if we could stay consistent and do the same I think that would be best. 
<img width="1792" alt="Screen Shot 2020-08-11 at 5 14 40 PM" src="https://user-images.githubusercontent.com/40238481/89949592-26f13080-dbf6-11ea-88b7-fdb5fb22ec1f.png">

## Potential future enhancements

> * In future, we may want to explore allowing users to set one of their saved searches to be the default view on the Search page so it is already selected upon page load.

Reviewer 3: ```suggestion
* In the future, we may want to explore allowing users to set one of their saved searches to be the default view on the Search page so it is already selected upon page load.
```
